lib/MusicBrainz/Server/Controller/WS/js.pm

```diff
@@ -49,6 +49,13 @@ sub medium : Chained('root') PathPart Args(1) {
     my ($self, $c, $id) = @_;
 
     my $medium = $c->model('Medium')->get_by_id($id);
+
+    unless ($medium) {
+        $c->stash->{error} = "No medium found with this ID.";
+        $c->detach('bad_req');
+        return;
```

Comment:
Technically the return is redundant since detach dies (i.e. never returns to the current action).